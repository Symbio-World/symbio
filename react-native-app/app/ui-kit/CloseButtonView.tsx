import * as React from 'react'
import { t } from 'react-native-tailwindcss'
import { CloseButton } from './CloseButton'
import { View } from 'react-native'

type CloseButtonViewProps = {
  onPress?: () => void
}

export const CloseButtonView: React.FC<CloseButtonViewProps> = ({
  onPress,
}) => (
  <>
    <View style={[t.h40]} />
    <View style={[t.absolute, t.bottom0, t.wFull, t.itemsCenter, t.pB5]}>
      <CloseButton onPress={onPress} />
    </View>
  </>
)
