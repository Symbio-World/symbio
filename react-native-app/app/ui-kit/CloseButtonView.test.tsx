import * as React from 'react'
import { render } from 'react-native-testing-library'
import { CloseButtonView } from './CloseButtonView'

describe('CloseButtonView', () => {
  it('renders correctly', () => {
    const handleClose = jest.fn()
    const { toJSON } = render(<CloseButtonView onClose={handleClose} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
