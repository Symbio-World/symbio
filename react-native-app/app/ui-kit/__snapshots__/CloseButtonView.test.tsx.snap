// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "height": 160,
        },
      ]
    }
  />,
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "bottom": 0,
        },
        Object {
          "width": "100%",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "paddingBottom": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderWidth": 2,
            },
            Object {
              "borderColor": "#B35340",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "width": 64,
            },
            Object {
              "height": 64,
            },
            Object {
              "borderRadius": 9999,
            },
            Object {
              "padding": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 20,
              },
              Object {
                "height": 20,
              },
            ]
          }
        >
          <SvgMock
            height="100%"
            width="100%"
          />
        </View>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "fontSize": 20,
              },
              Object {
                "color": "#B35340",
              },
            ]
          }
        >
          Close
        </Text>
      </View>
    </View>
  </View>,
]
`;
